Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: renaming libraries and paths so that builds for Qt5 and Qt6 are possible
----

--- a/qwtconfig.pri
+++ b/qwtconfig.pri
@@ -42,7 +42,7 @@
 # runtime environment of designer/creator.
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/lib/qt/plugins/designer
+QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/lib/qt$${QT_MAJOR_VERSION}/plugins/designer
 
 # linux distributors often organize the Qt installation
 # their way and QT_INSTALL_PREFIX doesn't offer a good
@@ -63,7 +63,7 @@
 # with every Qt upgrade.
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/lib/qt5/mkspecs/features
+QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/lib/qt$${QT_MAJOR_VERSION}/mkspecs/features
 # QWT_INSTALL_FEATURES  = $$[QT_INSTALL_PREFIX]/features
 
 ######################################################################
--- a/src/src.pro
+++ b/src/src.pro
@@ -17,7 +17,7 @@
 QWT_OUT_ROOT = $${OUT_PWD}/..
 
 TEMPLATE          = lib
-TARGET            = $$qwtLibraryTarget(qwt)
+TARGET            = $$qwtLibraryTarget(qwt-qt$${QT_MAJOR_VERSION})
 
 DESTDIR           = $${QWT_OUT_ROOT}/lib
 
@@ -31,7 +31,7 @@
     
             # we increase the SONAME for every minor number
 
-            QWT_SONAME=libqwt.so.$${VER_MAJ}.$${VER_MIN}
+            QWT_SONAME=libqwt-qt$${QT_MAJOR_VERSION}.so.$${VER_MAJ}.$${VER_MIN}
             QMAKE_LFLAGS *= $${QMAKE_LFLAGS_SONAME}$${QWT_SONAME}
             QMAKE_LFLAGS_SONAME=
         }
--- a/qwt.prf
+++ b/qwt.prf
@@ -44,4 +44,4 @@
 }   
 
 # QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
-qwtAddLibrary($${QWT_INSTALL_LIBS}, qwt)
+qwtAddLibrary($${QWT_INSTALL_LIBS}, qwt-qt$${QT_MAJOR_VERSION})
--- a/designer/designer.pro
+++ b/designer/designer.pro
@@ -98,7 +98,7 @@
         # compile the path for finding the Qwt library
         # into the plugin. Not supported on Windows !
 
-        qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+        qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt-qt$${QT_MAJOR_VERSION})
 
         contains(QWT_CONFIG, QwtDll) {
 
--- a/examples/examples.pri
+++ b/examples/examples.pri
@@ -34,7 +34,7 @@
 }
 
 QMAKE_RPATHDIR *= $${QWT_OUT_ROOT}/lib
-qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt-qt$${QT_MAJOR_VERSION})
 
 greaterThan(QT_MAJOR_VERSION, 4) {
 
--- a/playground/playground.pri
+++ b/playground/playground.pri
@@ -35,7 +35,7 @@
 
 
 QMAKE_RPATHDIR *= $${QWT_ROOT}/lib
-qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt-qt$${QT_MAJOR_VERSION})
 
 greaterThan(QT_MAJOR_VERSION, 4) {
 
--- a/tests/tests.pri
+++ b/tests/tests.pri
@@ -39,7 +39,7 @@
 
 
 QMAKE_RPATHDIR *= $${QWT_ROOT}/lib
-qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt-qt$${QT_MAJOR_VERSION})
 
 greaterThan(QT_MAJOR_VERSION, 4) {
 
